pkg/option/config.go

```diff
@@ -2850,6 +2850,10 @@ func (c *DaemonConfig) validateIPv6NAT46x64CIDR() error {
 
 // Validate validates the daemon configuration
 func (c *DaemonConfig) Validate(vp *viper.Viper) error {
+	if err := c.validatePreviousConfig(c.StateDir); err != nil {
+		return err
+	}
```

Comment:
AFAICT, this validation is affected by two possible drawbacks:
* It introduces a backward-compatibility requirement on the `DaemonConfig` struct. Let say for instance that at a certain point we had a `Foo` field of type string, which is then switched to be bool. The unmarshalling will then always fail because the two types are incompatible. I'm not sure how likely such a change could be, or whether others could trigger the same issue, but still worth mentioning.
* We currently store only the `DaemonConfig` configurations, which means that logic will no longer work after the refactoring has been completed, and the parameter has been removed from that struct.

That said, I'm still personally not fully convinced that the complexity introduced by this (or equivalent) validations is actually worth, considering that we don't do anything similar for a range of other functionalities (e.g., switching from tunnel to direct routing) which can equivalently disrupt existing connections.

Reply:
dropped :+1: